src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -249,7 +249,12 @@ private void SetUrl()
                 }
 
                 EventSourceTrace("Url: {0}", requestUri);
-                SetCurlOption(CURLoption.CURLOPT_URL, requestUri.AbsoluteUri);
+                string idnHost = requestUri.IdnHost;
```

Comment:
This PR is adding new overhead: there's at least an allocation for the IdnHost string plus another allocation (I assume) of the Host string, and the strings are being compared, which happens regardless of whether the host needs to be replaced. I wonder if it'd be cheaper to check if requestUri.AbsoluteUri contains any non-ASCII characters, and only do the host swap if that's the case. That way, the common case of all-ASCII uris don't have to allocate those additional Host/IdnHost strings.